operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -1,263 +1,379 @@
-          - default: memberOf
-          - type: string
-        * ldapRoleMapperUseRealmRolesMapping: If true then LDAP role mappings will be mapped to realm role mappings in Keycloak, otherwise the LDAP role mappings will be mapped to client role mappings.
-          - default: false
-          - type: string
-          - allowed values: true or false
-        * ldapRoleMapperClientId: If ldapRoleMapperUseRealmRolesMapping is false then this is the client ID to apply the roles to.
-          - default: shasta
-          - type: string
-        ```
 
-    2.  Encrypt the static values in the customizations.yaml file after making changes.
+### Procedure
 
+1. Prepare to customize the customizations.yaml file.
+
+   If the customizations.yaml file is managed in an external Git repository (as recommended), then clone a local working tree. Replace the `<URL>` value in the following command before running it.
+
+   ```bash
+   ncn-m001# git clone <URL> /root/site-init
+   ncn-m001# cd /root/site-init
+   ```
+
+   If there is not a backup of site-init, perform the following steps to create a new one using the values stored in the Kubernetes cluster.
+
+   1. Create a new site-init directory using from the CSM tarball.
+
+      ```bash
+      ncn-m001# cp -r ${CSM_DISTDIR}/shasta-cfg /root/site-init
+      ncn-m001# cd /root/site-init
+      ```
+  
+   1. Extract customizations.yaml from the site-init secret.
+
+      ```bash
+      ncn-m001# kubectl -n loftsman get secret site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d - > customizations.yaml
+      ```
+
+   1. Extract the certificate and key used to create the sealed secrets.
+
+      ```bash
+      ncn-m001# mkdir certs
+      ncn-m001# kubectl -n kube-system get secret sealed-secrets-key -o jsonpath='{.data.tls\.crt}' | base64 -d - > certs/sealed_secrets.crt
+      ncn-m001# kubectl -n kube-system get secret sealed-secrets-key -o jsonpath='{.data.tls\.key}' | base64 -d - > certs/sealed_secrets.key
+      ```
+
+  > **NOTE:** All subsequent steps of this procedure should be performed within the `/root/site-init` directory created in this step.
+
+2. Generate Sealed Secrets.
+   
+   Secrets are stored in `customizations.yaml` as `SealedSecret` resources 
+   (encrypted secrets), which are deployed by specific charts and decrypted by the
+   Sealed Secrets operator. But first, those Secrets must be seeded generated and
+   encrypted.
+
+   The LDAP server CA certificate goes into SealedSecret.
+
+   1. Load the `zeromq` container image required by Sealed Secret Generators:
+
+      > **NOTE:** A properly configured Docker or Podman environment is required.
+
+      ```bash
+      linux# /mnt/pitdata/${CSM_RELEASE}/hack/load-container-image.sh dtr.dev.cray.com/zeromq/zeromq:v4.0.5
+      ```
+
+   2. Re-encrypt existing secrets:
+
+      ```bash
+      linux# /mnt/pitdata/prep/site-init/utils/secrets-reencrypt.sh /mnt/pitdata/prep/site-init/customizations.yaml \
```

Comment:
at line 36 we created a customizations.yaml in /root/site-init , and at line 47 we were tole to perform all steps in the /root/site-init directory. Now this is using /mnt/pitdata/prep/site-init/customizations.yaml and not /root/site-init/customizations.yaml (or just customizations.yaml since I'm in that directory) ... seems wrong. Why 2 customization.yaml files?

Reply:
Step 2 was pulled in from another procedure to avoid having to jump back and forth to complete this procedure. I can change it to be just customizations.yaml for this procedure.